server/node_modules

```diff
@@ -0,0 +1 @@
+/npm/node_modules
```

Comment:
That's true, but the purpose of this is to keep all of the project's dependencies inside of the container across all environments. Installing, upgrading, or removing dependencies can happen either by:

- Modifying `package.json` and rebuilding the container image
- Dropping into the container the installing them manually (with `docker exec`).

Running `npm install` locally seems like the most convenient approach, but later things like inconsistencies between Node and NPM versions sneak in. Maybe the NPM install works locally, but then fails inside the container because the version you're running locally is newer.

I think I'd want to try and work on making the containerized workflow smoother for you vs. gutting the symlink. The symlink is basically a minor hack to trick NPM into working with `Dockerfile` `COPY` operations **and** volumes (`docker run --volume ...`). It is the most elegant hack I've found thus far.